src/transformers/modelcard.py

```diff
@@ -463,6 +472,20 @@ def from_trainer(
         dataset=None,
         dataset_args=None,
     ):
+        # Infer default from dataset
+        one_dataset = trainer.train_dataset if trainer.train_dataset is not None else trainer.eval_dataset
+        if dataset_tags is None and is_hf_dataset(one_dataset):
+            default_tag = one_dataset.builder_name
+            # Those are not real datasets from the Hub so we exclude them.
+            if default_tag not in ["csv", "json", "text"]:
```

Comment:
@lhoestq Do you see any other value to exclude or a better check? Similarly, should I check something below for the `config_name` before taking this as a dataset arg?

Reply:
You can add `pandas` and `parquet` as well.
(parquet is currently being added)

For the config name you can also ignore it if the `builder_name` is csv/json/text/pandas/parquet since it's going to be a hash based on the source files that are used.

Reply:
Thanks!